### ISSUE TITLE
(Some) Blocked Tabs close randomly or upon changes to plugin

### ISSUE THREAD

**Issue by rinrinrun** [proginosko/LeechBlockNG#501]
Hello,

like the title says, I have observed the problem that blocked tabs will close without me actively closing them.

Originally I saw them close, as far as I could tell, randomly, especially when they had been blocked for a while, at least a few weeks I'd say. Not all blocked tabs would close.

However, recently I have been dis/enabling the plugin/it's ability to run in private mode occasionally and sometimes only a few of the very newest blocked tabs would close immediately upon the change. 

The blocked tabs are in both cases still in the history and can be seen in the latest closed tabs and therefore reopened. The problem, especially with a large number of opened tabs, has really been noticing that they are gone before closing more tabs to reopen them. 

I have observed this on Firefox for Android on both my tablet and phone, as I don't use Leechblock otherwise.
FF version: 131.0.3
Leechblock version: 1.6.5

Do let me know if there is more information I can give you to help narrow the cause of this down

**Comment by proginosko** [comment]
Are you talking about the default blocking page? If so, that's a built-in extension page, and Firefox will always close those whenever the extension is reloaded internally. That would happen if you disable/re-enable the plugin or if you start a private browsing session (Firefox runs the extension as a separate process for private tabs). As it stands, there's little I can do about it, since it's not within the extension's control to determine when it is loaded/reloaded by the browser.

**Comment by rinrinrun** [comment]
yes, exactly. that's unfortunate. do you have any idea why it's so inconsistent(only happening sometimes and not closing all blocked tabs)?

Does Firefox do the reloading randomly as well? because in the random instances of the blocked tabs closing I'm sure that I didn't start private browsing or dis/enabled the plugin. 

thank you for the fast reply

**Comment by proginosko** [comment]
Sorry, I don't have a good theory about it. I don't use Firefox for Android. (I test the basic functionality of the extension on an Android emulator.)

One thing you could try: check that [this setting](https://support.mozilla.org/en-US/kb/auto-close-open-tabs-firefox-android) is set to "Never".

Is it happening to both private and non-private tabs?

**Comment by rinrinrun** [comment]
okay :(. I set that setting to never as soon as I install Firefox because I don't want my tabs to auto close. 

I can't comment on private tabs as I rarely use them and if I do they usually don't get blocked.
